packages/eslint-plugin-office-addins/src/utils.ts

```diff
@@ -173,3 +178,89 @@ export function isLoadFunction(node: TSESTree.MemberExpression): boolean {
     node.property.name === "load"
   );
 }
+
+export function isContextSyncIdentifier(node: TSESTree.Identifier): boolean {
+  return (
+    node.name === "context" &&
+    node.parent?.type === TSESTree.AST_NODE_TYPES.MemberExpression &&
+    node.parent?.parent?.type === TSESTree.AST_NODE_TYPES.CallExpression &&
+    node.parent?.property.type === TSESTree.AST_NODE_TYPES.Identifier &&
+    node.parent?.property.name === "sync"
+  );
+}
+
+export type OfficeApiReference = {
+  operation: "Read" | "Load" | "Write" | "Sync";
+  reference: Reference;
+};
+
+export function isLoadReference(node: TSESTree.Identifier) {
+  return (
+    node.parent &&
+    node.parent.type === TSESTree.AST_NODE_TYPES.MemberExpression &&
+    isLoadFunction(node.parent)
+  );
+}
+
+let proxyVariables: Set<Variable>;
+let apiReferences: OfficeApiReference[];
+export function findReferences(scope: Scope): OfficeApiReference[] {
+  proxyVariables = new Set<Variable>();
+  apiReferences = [];
+  findReferencesAuxiliar(scope);
+  return apiReferences;
+}
+
+function findReferencesAuxiliar(scope: Scope): void {
```

Comment:
Why the separate methods?

Reply:
The "normal" method is only needed to initialize the global variables. By doing this, we can ensure that the function will work across multiple calls.
Without this separation, if a user would call this function twice, the second call would have wrong results (given the global variables were not empty).

Reply:
Yes . . . but why is a wrapper method needed just to initialize global variables . . . why can't it be in the same method?

Reply:
Like, the first call on the function needs to be different than the rest, given the necessity of initializing the global variables every time.
The wrapper method is kinda way I`ve seen used before to solve this kind of problem. Do you think of another way? I could pass an additional variable to the functions (to detect the first time the function gets called), but I think this gets messier.

Reply:
Ah yes . . . I forgot about the recursive call.  Makes sense now.

Note that I tried passing in the array when I was doing it . . . but that was part of the trouble I had (mentioned in another comment) where the array wasn't keeping all the values (something with the way javascript loops work).  The global variable was the way to keep everything straight.

Personally I don't like the use of "Aux" or "Auxiliary" in the method name.  I think findReferencesInScope or findOfficeApiReferencesInScope or something like that makes more sense.

Reply:
Sure, I just did the refactoring